airflow/include/tasks/extract/astro_docs.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+import logging
+from urllib.parse import urldefrag, urljoin
+
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup
+from weaviate.util import generate_uuid5
+
+base_url = "https://docs.astronomer.io/"
+
+
+def fetch_page_content(url: str) -> str:
+    """
+    Fetches the content of a given URL.
+
+    :param url: URL of the page to fetch.
+    :return: HTML content of the page.
+    """
+    try:
+        response = requests.get(url, headers={"User-Agent": "Mozilla/5.0"})
+        if response.status_code == 200:
+            return response.content
+    except requests.RequestException as e:
+        logging.error(f"Error fetching {url}: {e}")
+    return ""
+
+
+def extract_links(soup: BeautifulSoup, base_url: str) -> list[str]:
```

Comment:
for this method and also for scrape_page can you check if you can use https://github.com/astronomer/ask-astro/blob/main/airflow/include/tasks/extract/utils/html_url_extractor.py#L72

Reply:
It won't work because we need to ignore certain URLs as per logic in the code